src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpointImpl.java

```diff
@@ -0,0 +1,396 @@
+        WatchImpl.measure("join attributes", () -> joinedAttributes.set(joinAttributes(typeAttributes.get(), attributes.get())));
+
+        ObjectWrapper<JsonObject> jsonEntity = new ObjectWrapper<>();
+        WatchImpl.measure("get entity json", () -> jsonEntity.set(getEntityJson(entity, joinedAttributes.get())));
+
+        return serializer.toJson(jsonEntity.get());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getChildEntities(Request request, Response response) {
+        long entityId = getEntityId(request);
+        long entityTypeId = getEntityTypeId(request);
+        List<Entity> childEntities = PersistenceAdapterImpl.getInstance().getEntities(entityId, entityTypeId);
+        List<TypeAttribute> typeAttributesToInclude = getAttributesTypeIdsToInclude(request);
+
+		ObjectWrapper<JsonArray> jsonEntities = new ObjectWrapper<>();
+		WatchImpl.measure("getEntitiesJson", () -> jsonEntities.set(getEntitiesJson(childEntities, typeAttributesToInclude)));
+
+        return serializer.toJson(jsonEntities.get());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventTypesOfEntity(Request request, Response response) {
+        long entityId = getEntityId(request);
+        boolean includeChildEvents = getIncludeChildEvents(request);
+
+        EntityHierarchyNode entityNode = HierarchyBuilderImpl.getInstance().getEntityHierarchyRootNode().findChildEntity(entityId);
+
+        if (entityNode == null) {
+        	halt(HttpStatusCodes.NOT_FOUND, "cannot find entity in hierarchy");
+		}
+
+		List<Long> entityIds = new ArrayList<>();
+
+        if (includeChildEvents) {
+		 	entityIds.addAll(entityNode.getChildIds());
+		} else {
+        	entityIds.add(entityId);
+		}
+
+		List<EventType> eventTypes = PersistenceAdapterImpl.getInstance()
+				.getEventTypes(entityIds.toArray(new Long[entityIds.size()]));
+
+        JsonArray eventTypesJson = getEventTypesJson(eventTypes);
+
+        return serializer.toJson(eventTypesJson);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventsOfEntity(Request request, Response response) {
+        long entityId = getEntityId(request);
+        long eventTypeId = getEntityTypeId(request);
+        boolean includeChildEvents = getIncludeChildEvents(request);
+        int startIndex = getStartIndex(request);
+        int endIndex = getEndIndex(request);
+
+        if (endIndex < startIndex) {
+        	int temp = endIndex;
+        	endIndex = startIndex;
+        	startIndex = temp;
+		}
+
+		EntityHierarchyNode entityNode = HierarchyBuilderImpl.getInstance().getEntityHierarchyRootNode().findChildEntity(entityId);
+
+        if (entityNode == null) {
+        	halt(HttpStatusCodes.NOT_FOUND, "cannot find entity in hierarchy");
+		}
+
+		List<Long> entityIds = new ArrayList<>();
+
+		if (includeChildEvents) {
+			entityIds.addAll(entityNode.getChildIds());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'entityNode' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)